_posts/2023-03-15-Automate-Software-Workflows-with-Github-Actions.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Tip of the Week: Automate Software Workflows with Github Actions"
+author: Dave Bunten
+member: dave-bunten
+tags:
+  - tip-of-the-week
+  - software
+  - github
+  - workflow
+  - github-actions
+  - continuous-integration
+---
+
+# Tip of the Week: Automate Software Workflows with Github Actions
+
+> Each week we seek to provide a software tip of the week geared towards helping you achieve your software goals. Views expressed in the content belong to the content creators and not the organization, its affiliates, or employees. If you have any software questions or suggestions for an upcoming tip of the week, please don’t hesitate to reach out to #software-engineering on Slack or email DBMISoftwareEngineering at olucdenver.onmicrosoft.com
+
+There are many routine tasks which can be automated to help save time and increase reproducibility in software development. Github Actions provides one way to accomplish these tasks using code-based workflows and related workflow implementations. This type of automation is commonly used to perform tests, builds (preparing for the delivery of the code), or delivery itself (sending the code or related artifacts where they will be used).
+
+__TLDR (too long, didn't read);__
+Use [Github Actions](https://docs.github.com/en/actions) to perform [continuous integration](https://en.wikipedia.org/wiki/Continuous_integration) work automatically by leveraging [Github's workflow specification](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions). You can test these workflows with [Act](https://github.com/nektos/act), which can enhance development with this feature of Github. Consider making use of ["write once, run anywhere" (WORA)](https://en.wikipedia.org/wiki/Write_once,_run_anywhere) and [Dagger](https://docs.dagger.io/) in conjunction with Github Actions to enable reproducible workflows for your software projects.
+
+## Workflows in Software
+
+<pre class="mermaid">
+flowchart LR
+  start((start)) --> action
+  action["action(s)"] --> en((end))
+  style start fill:#6EE7B7
+  style en fill:#FCA5A5
+</pre>
+<script type="module">
+  import mermaid from 'https://unpkg.com/mermaid@9/dist/mermaid.esm.min.mjs';
+  mermaid.initialize({ startOnLoad: true });
+</script>
+
+_An example workflow._
+{:.center}
+
+[Workflows](https://en.wikipedia.org/wiki/Workflow) consist of sequenced activities used by various systems. Software development  workflows help accomplish work the same way each time using code by implementing what are commonly called "workflow engines". Generally, workflow engines are provided code which indicate beginnings (what triggers a workflow to begin), actions (work being performed in sequence), and an ending (where the workflow stops). There are [many workflow engines](https://s.apache.org/existing-workflow-systems), including some which help accomplish work alongside version control.
```

Comment:
Just a minor tidy-up: the sentence had gotten a bit long, and you mention in the next sentence that you provide code to the workflow engine to define the workflow, anyway. I also changed "implementing" to "using"; I can see how you could loosely interpret "implementing" as putting anything in place (i.e., a workflow engine), but I think people often interpret "implementing" as writing code.

```suggestion
[Workflows](https://en.wikipedia.org/wiki/Workflow) consist of sequenced activities used by various systems. Software development  workflows help accomplish work the same way each time by using what are commonly called "workflow engines". Generally, workflow engines are provided code which indicate beginnings (what triggers a workflow to begin), actions (work being performed in sequence), and an ending (where the workflow stops). There are [many workflow engines](https://s.apache.org/existing-workflow-systems), including some which help accomplish work alongside version control.
```

Reply:
Thanks, this sounds better!